idea-plugin/src/main/kotlin/arrow/meta/ide/MetaRegistrar.kt

```diff
@@ -0,0 +1,50 @@
+package arrow.meta.ide
+
+import com.intellij.ide.ApplicationInitializedListener
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.Application
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.impl.ProjectLifecycleListener
+import com.intellij.util.messages.Topic
+import org.jetbrains.kotlin.config.CompilerConfiguration
+
+class MetaRegistrar : ApplicationInitializedListener {
+  val LOG = Logger.getInstance("#arrow.AppRegistrar")
+
+  override fun componentsInitialized(): Unit {
+    LOG.info("componentsInitialized")
+    ApplicationManager.getApplication()?.let { app ->
+      LOG.info("subscribing meta registrars")
+      val start = System.currentTimeMillis()
+      app.register(ProjectLifecycleListener.TOPIC, metaProjectRegistrar) // Alternative use ProjectManagerListener.TOPIC
+      LOG.info("subscribing meta registrars took ${System.currentTimeMillis() - start}ms")
+    }
+    println("componentsInitialized")
+  }
+}
+
+private fun <A> Application.register(topic: Topic<A>, vararg listeners: A): Unit =
+  listeners.toList().forEach { messageBus.connect(this).subscribe(topic, it) }
+
+/**
+ * This extension registers a MetaPlugin for a given project.
+ */
+private val metaProjectRegistrar: ProjectLifecycleListener
```

Comment:
Defining `MetaRegistrar` as an ApplicationInitializationListener is vital because it provides the DSL an explicit endpoint to register application-level changes and project-level changes an early point in the ide lifecycle. 
In later PRs this will be more clear when we decompose the IdePlugin from the MetaPlugin
#446 and push those capabilities to `IdeInternalRegistry`. 
In the end, users may register an compiler-plugin as an`ProjectLifecycleListener` with the Meta dsl.

Reply:
Okay, that makes sense